File: app/src/main/java/com/riders/thelab/data/remote/livedata/WeatherLiveDataHelper.java
package com.riders.thelab.data.remote.livedata;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MediatorLiveData;

public class WeatherLiveDataHelper {

    private static WeatherLiveDataHelper liveDataHelper;
    private final MediatorLiveData<Integer> downloadPercent =
            new MediatorLiveData<>();
    private WeatherLiveDataHelper() {
    }

    synchronized public static WeatherLiveDataHelper getInstance() {
        if (liveDataHelper == null)
            liveDataHelper = new WeatherLiveDataHelper();
        return liveDataHelper;
    }

    public void updateDownloadPer(int percentage) {
        downloadPercent.postValue(percentage);
    }

    public LiveData<Integer> observePercentage() {
        return downloadPercent;
    }
}

File: app/src/main/java/com/riders/thelab/data/remote/livedata/WeatherLiveDataHelper.kt
package com.riders.thelab.data.remote.livedata

import androidx.lifecycle.LiveData
import androidx.lifecycle.MediatorLiveData

class WeatherLiveDataHelper {

    companion object {
        private var liveDataHelper: WeatherLiveDataHelper? = null

        @Synchronized
        fun getInstance(): WeatherLiveDataHelper? {
            if (liveDataHelper == null) liveDataHelper = WeatherLiveDataHelper()
            return liveDataHelper
        }
    }

    private val downloadPercent = MediatorLiveData<Int>()

    fun updateDownloadPer(percentage: Int) {
        downloadPercent.postValue(percentage)
    }

    fun observePercentage(): LiveData<Int> {
        return downloadPercent
    }
}